src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CallVariantsFromAlignedContigsSpark.java

```diff
@@ -0,0 +1,382 @@
+        programGroup = StructuralVariationSparkProgramGroup.class)
+public final class CallVariantsFromAlignedContigsSpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+    public static final Integer DEFAULT_MIN_ALIGNMENT_LENGTH = 50;
+
+    @Argument(doc = "URI of the output path", shortName = "outputPath",
+            fullName = "outputPath", optional = false)
+    private String outputPath;
+
+    @Argument(doc = "Input file of contig alignments", shortName = "inputAlignments",
+            fullName = "inputAlignments", optional = false)
+    private String inputAlignments;
+
+    @Argument(doc = "Input file of assembled contigs", shortName = "inputAssemblies",
+            fullName = "inputAssemblies", optional = false)
+    private String inputAssemblies;
+
+    @Argument(doc = "Minimum flanking alignment length", shortName = "minAlignLength",
+            fullName = "minAlignLength", optional = true)
+    private Integer minAlignLength = CallVariantsFromAlignedContigsSpark.DEFAULT_MIN_ALIGNMENT_LENGTH;
+
+    // This class requires a reference parameter in 2bit format (to broadcast) and a reference in FASTA format
+    // (to get a good sequence dictionary).
+    // todo: document this better
+    @Argument(doc = "FASTA formatted reference", shortName = "fastaReference",
+            fullName = "fastaReference", optional = false)
+    private String fastaReference;
+
+    @Override
+    public boolean requiresReference() {
+        return true;
+    }
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+        Broadcast<ReferenceMultiSource> broadcastReference = ctx.broadcast(getReference());
+
+        final JavaRDD<AlignmentRegion> inputAlignedContigs = ctx.textFile(inputAlignments).map(AlignAssembledContigsSpark::parseAlignedAssembledContigLine);
+
+        final JavaPairRDD<Tuple2<String, String>, Iterable<AlignmentRegion>> alignmentRegionsKeyedByBreakpointAndContig = inputAlignedContigs.mapToPair(alignmentRegion -> new Tuple2<>(new Tuple2<>(alignmentRegion.assemblyId, alignmentRegion.contigId), alignmentRegion)).groupByKey();
+
+        final JavaPairRDD<String, ContigsCollection> assemblyIdsToContigCollections = loadContigsCollectionKeyedByAssemblyId(ctx, inputAssemblies).cache();
+
+        final JavaPairRDD<Tuple2<String, String>, byte[]> contigSequences = assemblyIdsToContigCollections.flatMapToPair(assemblyIdAndContigsCollection -> {
+            final String assemblyId = assemblyIdAndContigsCollection._1;
+            final ContigsCollection contigsCollection = assemblyIdAndContigsCollection._2;
+            final List<Tuple2<Tuple2<String, String>, byte[]>> contigSequencesKeyedByAssemblyIdAndContigId =
+                    contigsCollection.getContents().stream().map(pair -> new Tuple2<>(new Tuple2<>(assemblyId, pair._1.toString()), pair._2.toString().getBytes())).collect(Collectors.toList());
+            return contigSequencesKeyedByAssemblyIdAndContigId;
```

Comment:
I don't think naming the list adds any clarity.
`ComplicatedType varName = expression;
return varName;`
could just be
`return expression;`